### PR TITLE
feat(*): rewrite commands to use autocomplete, restructure project to be easier to extend

### PR DESCRIPTION
**Pull request by almostSouji**
- Use autocomplete for /guide
- Use autocomplete for /discorddocs (former /ddocs)
- Use subcommands for /docs to prevent arguments right of autocomplete, the ac depends on
- Use native crypto module to verify discord payloads, removing dependency on third party packages
- Automaticall prefix the description of test commands
- Move handling logic into separate file per application command type
- Fully remove ephemeral selection fallback
- Move to use yarn as package manager
- Escape some MD in MDN titles `*` `||` 

This builds on #52 as a larger scope rewrite
This PR is currently running, so tested in production

### CODE REVIEW

**Review by SpaceEEC — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

File: src/functions/autocomplete/tagAutoComplete.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+export function tagAutoComplete(
+	res: Response,
+	options: APIApplicationCommandInteractionDataOption[],
+	tagCache: Collection<string, Tag>,
+): Response {
+	const { query } = transformInteraction<typeof TagCommand>(options);
+	const results: { name: string; value: string }[] = [];
+
+	// eslint-disable-next-line @typescript-eslint/prefer-optional-chain
+	if (query && query.length) {
```
Comment (vladfrangu) [inline src/functions/autocomplete/tagAutoComplete.ts:17]: Why don't you `query?.length`

File: src/handling/handleApplicationCommand.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+					castArgs.query,
+					EMOJI_ID_GUIDE,
+					castArgs.target,
+				);
+				break;
+			}
+			case 'invite': {
+				prepareResponse(
+					res,
+					`Add the discord.js interaction to your server: [(click here)](<https://discord.com/api/oauth2/authorize?client_id=${process
```
Comment (vladfrangu) [inline src/handling/handleApplicationCommand.ts:83]: Use the hyperlink util from /builders

File: src/index.ts [hunk 3]
```diff
@@ -1,191 +1,117 @@
...
+Doc.setGlobalOptions({
+	escapeMarkdownLinks: true,
+});
+
+export async function start() {
+	const mdnData = (await fetch(`${API_BASE_MDN}/en-US/search-index.json`)
+		.then((r) => r.json())
+		.catch(() => undefined)) as MDNIndexEntry[] | undefined;
+	if (mdnData) {
+		const mdnResult = mdnData;
```
Comment (vladfrangu) [inline src/index.ts:83]: This is a useless var

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/functions/autocomplete/tagAutoComplete.ts (hunk 1 above)
Comment (almostSouji) [reply src/functions/autocomplete/tagAutoComplete.ts:17]: legacy from the selection ephemeral, can't even be nullish anymore

File: src/index.ts (hunk 3 above)
Comment (almostSouji) [reply src/index.ts:83]: I think there used to be a typecast here, removed

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**